### DISCUSSION TITLE
Favorite AI For Debugging Quasar?

### DISCUSSION THREAD

**Discussion by vahost** [quasarframework/quasar#17025] [General - Components / Directives / etc]
I've never been good at programming. I gave up on it for a few years because I was so frustrated by it. However, the recent advent of A.I. as an aid to programmers made me give it another try.

So I resurrected an old Quasar project and tried working with Gemini and ChatGPT4 to try fix some old problems and some new ones, as well as to work on a new project.

A.I. is definitely helpful. I've been making strides with its assistance.

However, I still encounter issues that I find I cannot resolve, even with A.I.'s help. For instance, right now I'm having trouble with a 5-day calendar component. Despite my best efforts, i've been unable to resolve a problem where clicking from one 5 day range to the next jumps 7 days instead of 5 days.

I was wondering what A.I.'s you all use to assist you with your Quasar programming. Has anyone found one that is particularly good at debugging and resolving dynamic coding issues?

**Comment by vahost** [comment]
Wow, no one here has a favorite A.I. for programming? Really?

**Reply by bzuidgeest** [reply]
No, because AI won't magically solve your problems. If you cannot understand the code they provide and how it works, your a danger to an employer and yourself... You might write in major security issues. They also still hallucinate a lot. Creating functions using non-existing dependencies.

If I am really out of ideas I sometimes use one, but I never use the code they provide directly. I always rewrite it myself. I use chatGPT, for no other reason it was first. Never tried the others. Almost no use for them

That said, if it helps you progress and have some fun, they can be excellent tools and I don't have anything against their use. As long as it to increase knowledge and understanding, not to fill in for it. Always make sure you actually understand what they provide.

**Reply by vahost** [reply]
@bzuidgeest I think it's really fascinating that you transformed "I've never been good at programming. I gave up on it for a few years because I was so frustrated by it. " and  "Has anyone found one that is particularly good at debugging and resolving dynamic coding issues?" into "If you cannot understand the code they provide and how it works, your a danger to an employer and yourself... "

Never mind the fact that I am not employed as a programmer. Even without that fact, your reply made some wild assumptions. But I do thank you for replying at all.

**Comment by Dtsiantaris** [comment]
Generally speaking, AI is not really good at complex problems. 
For example the one you mention about the calendar might be a lot of things. 
You cannot just drop 500+ lines of code or multiple components/files and expect the AI to come up with a solution. 
When it comes to framework specific problems it's even worse. I would suggest [phind](https://www.phind.com/search?home=true). 
It can search the internet, look at doc's if you provide it with a link(ie quasar docs) and even ask you questions to clarify the nature of your prompt/problem.
AI is indeed a great tool but just don't rely too much on it.

**Reply by vahost** [reply]
Thank you for suggesting phind. I'm definitely going to check that out.

We're not at the stage where AI can create a whole project from soup-to-nuts ... not yet, anyway. But that day is coming sooner than you might think.

I believe that AI is at the same stage, metaphorically speaking, as Google was in 1999. It took Google about 10-15 years to become the Google of today. I expect AI will develop a lot faster and in a lot more ways in the next 3-5 years than Google ever had to do in 15 years. We are within shouting distance of AI passing the Turing Test for most people (not programmers ... that will take longer).
